### PR TITLE
pcolorfast simplifications.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -6213,7 +6213,7 @@ def pcolorfast(self, *args, alpha=None, norm=None, cmap=None, vmin=None,
                 "'norm' must be an instance of 'mcolors.Normalize'")
 
         C = args[-1]
-        nr, nc = C.shape
+        nr, nc = np.shape(C)
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: (supports C being a nested list instead of an array)

**Comment by timhoffm** [comment]
This wouldn't do any harm even if we decide to deprecate `pcolorfast`.

**Comment by ImportanceOfBeingErnest** [comment]
I would be supportive of keeping pcolorfast, even if it is in general of course really bad style to have the return type of the function depend on some (possibly complex) interplay of input parameters. That being said, `pcolorfast` is almost completely untested. So if a test could be added to check the return type and limits depending on all combinations of input options that would really help.

**Comment by anntzer** [comment]
See https://github.com/matplotlib/matplotlib/pull/13434 for some more testing of pcolorfast.

**Comment by jklymak** [comment]
My issue w/ `pcolorfast` is that it is redundant with `pcolormesh` and `imshow`.  Again, asside from the API being different, it doesn't *do* anything that those two methods don't do.

**Comment by anntzer** [comment]
Let's keep that discussion in #13328, but in that thread I made the point that I disagree with the assessment that the two methods "provide" similar functionalities.  Using imshow() to draw an array *so that a certain data entry ends up centered at a certain x, y coordinate* is extremely awkward (in particular because of the complex interplay between the extent and the origin kwargs).  It is true that anything pcolorfast does can be translated into an imshow/pcolorfast/pcolormesh call, but that's like saying that we don't need `pie()` because we could get the same thing by manually drawing slices on a polar plot with `bar()`.

**Comment by ImportanceOfBeingErnest** [comment]
What this PR does is to simplify the internal code of an existing, and for some users viable function. It doesn't add anything, and it doesn't remove anything. So I wouldn't spend too much time with it. Only thing is that such "simplifications" are always a good opportunity to add lacking tests, so if possible, I would like to see #13434 be merged first.

**Comment by timhoffm** [comment]
@ImportanceOfBeingErnest  #13434 is merged.

**Comment by anntzer** [comment]
@ImportanceOfBeingErnest do you want to get this in? :)

**Pull request merged**